app/views/course/announcements/_form.html.slim

```diff
@@ -0,0 +1,12 @@
+= simple_form_for [@course, @announcement],
+                    url: (@announcement.new_record? ?  course_announcements_url(@course) : \
+                         course_announcement_url(@course, @announcement)) do |f|
```

Comment:
So I think we should do this by inheritance ? That is override self.model_name in the parent model.  

Similar to what we do in controllers